### ISSUE TITLE
dropbox fails to start

### ISSUE THREAD

**Issue by phunehehe** [NixOS/nixpkgs#11728]
```
Couldn't start Dropbox.
This is usually because of a permissions error. Storing your home folder on a network share can also cause an error.

Get more help at https://www.dropbox.com/c/help/permissions_error

Please contact Dropbox support with the following info for help:

/tmp/dropbox_error9KrtSh.txt
```

The log file is https://gist.github.com/phunehehe/d9a053d156f5054aecc9

I suspect this has already been fixed in master, so I'm trying to reproduce, but downloading from https://dl.dropboxusercontent.com/u/17/dropbox-lnx.x86_64-3.12.4.tar.gz is taking a while. If it works, https://github.com/NixOS/nixpkgs/compare/release-15.09...phunehehe:dropbox-15.09 might be the right thing to do. I'll be back when I get more info.

**Comment by phunehehe** [comment]
Never mind my suspicion. Dropbox doesn't work from master either (same error). Any suggestion?

Pinging @ttuegel @bendlas since you guys seem to be somewhat involved with pkgs/applications/networking/dropbox/default.nix :)

**Comment by jagajaga** [comment]
Works nice for me..

**Comment by phunehehe** [comment]
@jagajaga do you use nixos or do you use standalone nixpkgs (I use standalone nixpkgs)?

**Comment by jagajaga** [comment]
I use Nixos, but I use dropbox installed via `nix-env`.

**Comment by bendlas** [comment]
@phunehehe your logfile says `ImportError: libGL.so.1: cannot open shared object file: No such file or directory`, coming from PyQT. You should be able to fix this, by adding the directory containing your libGL.so.1 to LD_LIBRARY_PATH

ping PyQT maintainers @svanderburg @iyzsong
- why does PyQT unconditionally load libGL (I don't think dropbox ui uses it), can we get this fixed upstream?
- can or should we do anything for nixpkgs users, with no libGL on their LD_LIBRARY_PATH?

**Comment by phunehehe** [comment]
@bendlas thanks, I can confirm `LD_LIBRARY_PATH=/home/phunehehe/.nix-profile/lib dropbox` works! Am I supposed to add this `LD_LIBRARY_PATH=/home/phunehehe/.nix-profile/lib` thing to my shell profile though?

**Comment by bendlas** [comment]
@phunehehe no, LD_LIBRARY_PATH overrides the system-builtin path and could lead to unforseen problems with programs installed from your regular system
For now, I'd recommend a wrapper in your bin folder:
```sh
#!/bin/sh
LD_LIBRARY_PATH=... exec dropbox "$@"
```

**Comment by bendlas** [comment]
@phunehehe if there is a libGL installed on your system, but not found by nixpkgs' dropbox, you can add a LD_LIBRARY_PATH="/usr/lib/.." to you shell profile (which would only override system libs with themselves)

**Comment by phunehehe** [comment]
@bendlas I prefer to keep stuff in Nix, so I'm not gonna pull the file in via the other system :) Had to install `mesa` to get it in my Nix profile though. Maybe the Dropbox package should provide this wrapper then? That's probably easy to make.

**Comment by cleverca22** [comment]
under nixos, LD_LIBRARY_PATH=/run/opengl-driver/lib:/run/opengl-driver-32/lib

which then contains the mesa(or closed-source variant) that best fits your xserver driver

**Comment by bendlas** [comment]
@phunehehe I suppose that, since mesa is already a transitive dependency of dropbox (via qt), the wrapper could be updated. I think that this really should be fixed in (Py)QT, though.

**Comment by bendlas** [comment]
@phunehehe I've updated the dropbox wrapper, to add libGL to the path.
stable: https://github.com/NixOS/nixpkgs/pull/11836
master: https://github.com/NixOS/nixpkgs/pull/11835

**Comment by phunehehe** [comment]
Thanks @bendlas I'm updating right now!

**Comment by bendlas** [comment]
heh, seem this thread is still being notified, when I rebase my patch queue. I'm dropping that commit now, since it's in stable. For master, this is fixed by 9f62af0
